sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/ModelTests.groovy

```diff
@@ -0,0 +1,162 @@
+package com.azure.storage.file.datalake
+
+import com.azure.storage.file.datalake.models.AccessControlType
+import com.azure.storage.file.datalake.models.PathAccessControlEntry
+import com.azure.storage.file.datalake.models.PathPermissions
+import com.azure.storage.file.datalake.models.RolePermissions
+import spock.lang.Unroll
+
+class ModelTests extends APISpec{
+
+    def "RolePermissions creation"() {
+        when:
+        def permissions = new RolePermissions(true, true, true)
+        def other = new RolePermissions(permissions)
+
+        then:
+        permissions.read()
+        permissions.execute()
+        permissions.write()
+
+        other.read()
+        other.execute()
+        other.write()
+
+        permissions.equals(other)
+    }
+
+    @Unroll
+    def "RolePermissions parse Octal"() {
+        expect:
+        RolePermissions.parseOctal(octal as int).toSymbolic() == permission
+
+        where:
+        octal | permission
+        1     | "--x"
+        2     | "-w-"
+        4     | "r--"
+    }
+
+    @Unroll
+    def "RolePermissions parse Symbolic"() {
+        expect:
+        RolePermissions.parseSymbolic(symbol, false).toOctal() == permission
+
+        where:
+        symbol | permission
+        "--x"  | "1"
+        "-w-"  | "2"
+        "r--"  | "4"
+    }
+
+    def "PathPermissions parse Symbolic"() {
+        when:
+        def permissions = PathPermissions.parseSymbolic("r---w---t+")
+        def owner = RolePermissions.parseOctal(4)
+        def group = RolePermissions.parseOctal(2)
+        def other = RolePermissions.parseOctal(1)
+
+        then:
+        permissions.owner() == owner
+        permissions.group() == group
+        permissions.other() == other
+        permissions.stickyBit()
+        permissions.extendedInfoInAcl()
+    }
+
+    def "PathPermissions create"() {
+        when:
+        def permissions = PathPermissions.parseSymbolic("r---w---t+")
+        def otherPermission = new PathPermissions(permissions)
+        def owner = RolePermissions.parseOctal(4)
+        def group = RolePermissions.parseOctal(2)
+        def other = RolePermissions.parseOctal(1)
+
+        then:
+        otherPermission.owner() == owner
+        otherPermission.group() == group
+        otherPermission.other() == other
+        otherPermission.stickyBit()
+        otherPermission.extendedInfoInAcl()
+        permissions.equals(otherPermission)
+    }
+
+    @Unroll
+    def "PathPermissions parse"() {
+        when:
+        def permissions = PathPermissions.parseSymbolic(symbol)
+
+        then:
+        permissions.other().execute() == execute
+        permissions.other().read() == read
+        permissions.other().write() == write
+        permissions.stickyBit() == stickyBit
+        permissions.extendedInfoInAcl() == extendedInfoInAcl
+
+        where:
+        // These test the value of other
+        symbol      | execute | read | write | stickyBit | extendedInfoInAcl
```

Comment:
nit: spock style suggests using a || to separate inputs and expected outputs, so the column after symbol should be a column of ||